pytest_sftpserver/sftp/content_provider.py

```diff
@@ -50,28 +126,49 @@ def remove(self, path):
         else:
             try:
                 delattr(obj, name)
-                return True
             except (TypeError, AttributeError):
-                pass
-        return False
+                return False
+            else:
+                self._st_times.pop((path, name), None)
+                self._update_times(dirpath, dirname, [None, time.time()])
+                return True
 
     def list(self, path):
         obj = self._find_object_for_path(path)
+        if obj is not None:
+            dirpath, dirname = self._get_path_components(path)
+            self._update_times(dirpath, dirname, [time.time(), None])
+
         if isinstance(obj, dict):
             return obj.keys()
         elif isinstance(obj, (list, tuple)):
             return [str(i) for i in range(len(obj))]
         else:
             return [n for n in dir(obj) if not n.startswith("__")]
 
+    def recursive_list(self, path):
+        subpath, subname = self._get_path_components(path)
+        yield subpath if subpath != '/' else '', subname
+        if self.is_dir(path):
+            for name in self.list(path):
+                fullname = posixpath.join(path, name)
+                for subpath, subname in self.recursive_list(fullname):
+                    yield subpath, subname
+
     def is_dir(self, path):
-        return not isinstance(self.get(path), string_types + integer_types)
+        # Using _find_object_for_path to avoid attribute-setting
+        # that .get() does.
+        return not isinstance(self._find_object_for_path(path),
+                              string_types + integer_types)
 
     def get_size(self, path):
         try:
-            return len(self.get(path))
+            return len(self.get(path, atime_change=False))
         except TypeError:
-            return len(str(self.get(path)))
+            return len(str(self.get(path, atime_change=False)))
```

Comment:
Just noticed that this would be wrong in py3k in light of #15 (which doesn't fix this).